DevConsoleRunScheduledTaskTest is flaky

https://github.com/quarkusio/quarkus/runs/2115516770?check_suite_focus=true
```
[INFO] Running io.quarkus.scheduler.test.devconsole.DevConsoleRunScheduledTaskTest
Error:  WARNING: An illegal reflective access operation has occurred
Error:  WARNING: Illegal reflective access by org.codehaus.groovy.vmplugin.v9.Java9 (file:/C:/Users/runneradmin/.m2/repository/org/codehaus/groovy/groovy/3.0.7/groovy-3.0.7.jar) to constructor java.lang.AssertionError(java.lang.String)
Error:  WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.vmplugin.v9.Java9
Error:  WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
Error:  WARNING: All illegal access operations will be denied in a future release
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
            Powered by Quarkus 999-SNAPSHOT
2021-03-15 19:14:36,401 INFO  [io.quarkus] (Quarkus Main Thread) Quarkus 999-SNAPSHOT on JVM started in 2.820s. Listening on: http://localhost:8080
2021-03-15 19:14:36,402 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2021-03-15 19:14:36,402 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [cdi, scheduler]
2021-03-15 19:14:36,633 INFO  [io.quarkus] (Quarkus Main Thread) Quarkus stopped in 0.022s
Error:  Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 3.582 s <<< FAILURE! - in io.quarkus.scheduler.test.devconsole.DevConsoleRunScheduledTaskTest
Error:  io.quarkus.scheduler.test.devconsole.DevConsoleRunScheduledTaskTest.testInvokeScheduledTask  Time elapsed: 3.574 s  <<< FAILURE!
java.lang.AssertionError: 
1 expectation failed.
Response body doesn't match expectation.
Expected: "true"
  Actual: false

	[...]
	at io.quarkus.scheduler.test.devconsole.DevConsoleRunScheduledTaskTest.testInvokeScheduledTask(DevConsoleRunScheduledTaskTest.java:34)
```
Came up here: https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/New.20catalog.20API/near/230467384

/cc @stuartwdouglas

--- a/extensions/scheduler/deployment/src/test/java/io/quarkus/scheduler/test/devconsole/DevConsoleRunScheduledTaskBodyHandlerTest.java
+++ b/extensions/scheduler/deployment/src/test/java/io/quarkus/scheduler/test/devconsole/DevConsoleRunScheduledTaskBodyHandlerTest.java
@@ -19,10 +19,10 @@ public class DevConsoleRunScheduledTaskBodyHandlerTest {
     @Test
     public void testInvokeScheduledTask() {
         RestAssured.with()
-                .get("status")
+                .get("empty")
                 .then()
                 .statusCode(200)
-                .body(Matchers.equalTo("false"));
+                .body(Matchers.equalTo("true"));
         RestAssured.with().formParam("name", "io.quarkus.scheduler.test.devconsole.NeverRunTask#run")
                 .redirects().follow(false)
                 .post("q/dev/io.quarkus.quarkus-scheduler/schedules")
@@ -32,7 +32,7 @@ public class DevConsoleRunScheduledTaskBodyHandlerTest {
                 .get("status")
                 .then()
                 .statusCode(200)
-                .body(Matchers.equalTo("true"));
+                .body(Matchers.equalTo("task ran"));
     }
 
 }

--- a/extensions/scheduler/deployment/src/test/java/io/quarkus/scheduler/test/devconsole/DevConsoleRunScheduledTaskTest.java
+++ b/extensions/scheduler/deployment/src/test/java/io/quarkus/scheduler/test/devconsole/DevConsoleRunScheduledTaskTest.java
@@ -18,10 +18,10 @@ public class DevConsoleRunScheduledTaskTest {
     @Test
     public void testInvokeScheduledTask() {
         RestAssured.with()
-                .get("status")
+                .get("empty")
                 .then()
                 .statusCode(200)
-                .body(Matchers.equalTo("false"));
+                .body(Matchers.equalTo("true"));
         RestAssured.with().formParam("name", "io.quarkus.scheduler.test.devconsole.NeverRunTask#run")
                 .redirects().follow(false)
                 .post("q/dev/io.quarkus.quarkus-scheduler/schedules")
@@ -31,7 +31,7 @@ public class DevConsoleRunScheduledTaskTest {
                 .get("status")
                 .then()
                 .statusCode(200)
-                .body(Matchers.equalTo("true"));
+                .body(Matchers.equalTo("task ran"));
     }
 
 }

--- a/extensions/scheduler/deployment/src/test/java/io/quarkus/scheduler/test/devconsole/NeverRunTask.java
+++ b/extensions/scheduler/deployment/src/test/java/io/quarkus/scheduler/test/devconsole/NeverRunTask.java
@@ -1,5 +1,6 @@
 package io.quarkus.scheduler.test.devconsole;
 
+import java.util.concurrent.LinkedBlockingDeque;
 import java.util.concurrent.TimeUnit;
 
 import javax.enterprise.event.Observes;
@@ -11,20 +12,31 @@ import io.vertx.ext.web.RoutingContext;
 
 public class NeverRunTask {
 
-    private static volatile boolean run;
+    private static final LinkedBlockingDeque<String> queue = new LinkedBlockingDeque<>();
 
     public void setup(@Observes Router router) {
-        router.route("/status").handler(new Handler<RoutingContext>() {
+        router.route("/status").blockingHandler(new Handler<RoutingContext>() {
             @Override
             public void handle(RoutingContext event) {
-                event.response().end(Boolean.toString(run));
+                try {
+                    event.response().end(queue.poll(10, TimeUnit.SECONDS));
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                    event.fail(e);
+                }
+            }
+        });
+        router.route("/empty").handler(new Handler<RoutingContext>() {
+            @Override
+            public void handle(RoutingContext event) {
+                event.response().end(Boolean.toString(queue.isEmpty()));
             }
         });
     }
 
     @Scheduled(every = "2h", delay = 2, delayUnit = TimeUnit.HOURS)
     public void run() {
-        run = true;
+        queue.add("task ran");
     }
 
 }

